src/Registration/cReg/NiftiImageData.cpp

```diff
@@ -183,6 +192,40 @@ std::shared_ptr<nifti_image> NiftiImageData<dataType>::create_from_geom_info(con
     return _nifti_image;
 }
 
+template<class dataType>
+void NiftiImageData<dataType>::construct_NiftiImageData_from_complex_im_real_component(std::shared_ptr<NiftiImageData> &out_sptr, const std::shared_ptr<const ImageData> in_sptr)
+{
+    // Create image from input
+    out_sptr = std::make_shared<NiftiImageData<dataType> >(*in_sptr);
```

Comment:
slightly confused. Isn't this  copying the data already? It's calling the constructor... So what would it do if the input data is complex? (And if it isn't, then just calling the copy constructor would be fine)

Reply:
I create a clone to get the shape and metadata etc. Then correct the voxel values. I think it takes the constructor takes the magnitude, so then they get replaced with the real component.

Reply:
ok. seems wasteful, but let's just add a TODO. (Unless we happen to have a `get_uniform_copy` already (which we do for `STIRImageData`)